core/src/main/scala/org/apache/spark/shuffle/BlockStoreShuffleReader.scala

```diff
@@ -51,7 +59,10 @@ private[spark] class BlockStoreShuffleReader[K, C](
       // Note: we use getSizeAsMb when no suffix is provided for backwards compatibility
       SparkEnv.get.conf.getSizeAsMb("spark.reducer.maxSizeInFlight", "48m") * 1024 * 1024,
       SparkEnv.get.conf.getInt("spark.reducer.maxReqsInFlight", Int.MaxValue),
-      SparkEnv.get.conf.getBoolean("spark.shuffle.detectCorrupt", true))
+      SparkEnv.get.conf.getLong("spark.reducer.maxReqSizeShuffleToMem", 200 * 1024 * 1024),
```

Comment:
let's also put the off-heap size conf in config object near the `maxReqSizeShuffleToMem` , so that it's easy to see `maxReqSizeShuffleToMem` is smaller than off-heap size by default.